nixos/modules/virtualisation/amazon-init.nix

```diff
@@ -26,12 +26,14 @@ let
           printf "%s" "$channels" > /root/.nix-channels
           nix-channel --update
         fi
-
-        echo "setting configuration from EC2 user data"
-        cp "$userData" /etc/nixos/configuration.nix
-      else
-        echo "user data does not appear to be a Nix expression; ignoring"
-        exit
+        nix-instantiate --eval-only "$userData" &> /dev/null
+        if [ $? -eq 0 ]; then
+          echo "setting configuration from EC2 user data"
+          cp "$userData" /etc/nixos/configuration.nix
+        else
+          echo "user data does not appear to be a Nix expression; ignoring"
```

Comment:
You should probably change wording of this to indicate that the expression failed to evaluate.

Reply:
@veprbl  thanks for the review. Would `Failed to evaluate user data with nix-instantiate; ignoring` be a more suitable message.

Reply:
Ok, maybe, evaluate wasn't a right word here. How about "user data is not a valid Nix expression; ignoring"?

Reply:
@veprbl yeah thats wayy better.